feat: implement updatemany, deletemany, getmany

wesleygrimes opened the pull request:
Implemented new updateMany, deleteMany, getMany methods on service

impl #14

wesleygrimes committed:
feat: implement updatemany, deletemany, getmany

Implemented new updateMany, deleteMany, getMany methods on service

impl #14

cmwhited commented on lib/services/in-memory-db.service.ts:
```diff
@@ -110,13 +123,40 @@ export class InMemoryDBService<T extends InMemoryDBEntity> {
+      .filter(id => this.recordMap[id])
+      .map(id => {
+        return this.recordMap[id];
+      });
```
niceeeee. i like this impl.

cmwhited commented on lib/services/in-memory-db.service.ts:
```diff
@@ -110,13 +123,40 @@ export class InMemoryDBService<T extends InMemoryDBEntity> {
+      this.recordMap = {
+        ...remainder,
+      };
+    });
```
This is a bit interesting. It works, but I wonder the implications of it and if there is a different way to do it.

wesleygrimes committed:
feat: refactored *Many methods to use existing single methods so that logic is not repeated

Switched over to use delete keyword on object, also using existing singular methods to power many
methods
